### ISSUE TITLE
Use heading elements

### ISSUE THREAD

**Issue by hanshillen** [adaptlearning/adapt-contrib-core#235]
Headings are currently marked up as follows:

```html
<div class="js-heading is-incomplete">
    <div id="c-15-heading" class="js-heading-inner" role="heading" aria-level="0">
        <span class="aria-label">Incomplete. My title</span>
    </div>
</div>
<div class="component__title-inner text__title-inner" aria-hidden="true">My title</div>
```
So the actual heading is visually hidden, while the visible heading is programmatically hidden from assistive tech. This in itself is not  semantically incorrect, but the approach has a few side effects:

* Any assistive tech product, tool, extension, or user-defined style sheet that targets headings (by highlighting them or overriding their styles) won't be able to do so; the programmatically detectable heading is not visible (and its bounds do not match the bounds of the visible heading).
* Since the visible heading is the same selector regardless of its level, there's no easy selector available to style heading levels individually (e.g. to make an h2 bigger than an h3). Using classes like `component__title-inner` won't work, since that heading is not guaranteed to be the same level for every component. The `aria-level` attribute is on the invisible element, so not suitable for a selector either (unless you do something crazy like `.js-heading:has(>[aria-level="2"])+.component__title-inner)` which won't work anyway).
* It's unnecessarily complicated and redundant, without offering any benefits over regular headings.

In my theme I've changed the templates to use heading elements instead (using the ariaLevel value for the element name). Making them look like this:

```html
<div class="js-heading component__title-inner text__title-inner is-incomplete">
  <h3 id="c15-heading" class="js-heading-inner">     
    <span class="aria-label">
        Incomplete.
    </span>
    My title
  </h3>
</div>
```

I think it would be useful to use this approach in core as well, as it resolves the side effects listed above while preserving the semantics. 

Note: This is a feature request. not a bug. If the change is not feasible because it would break other plugins (like themes and pageLevelProgress), then this issue can be closed. The issue and associated PR can still be useful for others looking to do the same in their theme in the future.

**Comment by oliverfoster** [comment]
I've gone with unifying the headings but not using semantic h tags as it's more complicated to override the native h tag styling in a way that's backward compatible.

**Comment by oliverfoster** [comment]
@hanshillen could you have a look at the above prs and let me know what you think?
We can reconsider h tags at a later date if that's agreeable?

**Comment by hanshillen** [comment]
lgtm, the semantically exposed heading and the visual heading are the same now, and heading levels can be styled separately with the `[role=heading][aria-level=x]` or `.js-heading-inner[aria-level=x]` selectors.

**Comment by oliverfoster** [comment]
Cool. If you're happy would you be able to approve each of those prs? We need three approvals to get them merged, two from core members and one from anyone else.
